### PR TITLE
Add refined-pureconfig

### PR DESCRIPTION
**Pull request by vlovgr**
Adds a `refined-pureconfig` project containing a tiny integration with [pureconfig](https://github.com/melrief/pureconfig).  
I've tried to keep things as similar as possible to `refined-scodec` where possible.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -165,13 +165,14 @@ If you're using sbt, add the following to your build:
...
-  "eu.timepit" %% "refined-scalacheck" % "0.6.0" % "test" // optional
+  "eu.timepit" %% "refined-scalaz"     % "0.6.0",          // optional
+  "eu.timepit" %% "refined-scodec"     % "0.6.0",          // optional
+  "eu.timepit" %% "refined-scalacheck" % "0.6.0" % "test", // optional
+  "eu.timepit" %% "refined-pureconfig" % "0.6.0"           // optional
 )
 ```
 
-For Scala.js just replace `%%` with `%%%` above.
+For Scala.js just replace `%%` with `%%%` above (except for `refined-pureconfig` which is not available for Scala.js).
```
Comment (vlovgr) [inline README.md]: Issue for pureconfig scala.js support: https://github.com/melrief/pureconfig/issues/46

File: build.sbt [hunk 2]
```diff
@@ -167,7 +188,20 @@ lazy val submoduleSettings = Def.settings(
 )
 
 lazy val submoduleJvmSettings = Def.settings(
-  mimaPreviousArtifacts := Set(groupId %% moduleName.value % latestVersion),
+  mimaPreviousArtifacts := {
```
Comment (vlovgr) [inline build.sbt]: I couldn't get `mimaReportBinaryIssues` to work without these changes. (We could potentially remove this after a release with `refined-pureconfig` has been issued.)

**Comment by fthomas** [comment]
Awesome @vlovgr. Thanks for this early christmas present! :-)

File: build.sbt (hunk 2 above)
Comment (fthomas) [inline build.sbt]: :+1: Or if we leave it in, adding the next submodule will be easier.

File: contrib/pureconfig/jvm/src/main/scala/eu/timepit/refined/pureconfig/error/PredicateFailedException.scala [hunk 3]
```diff
@@ -0,0 +1,4 @@
+package eu.timepit.refined.pureconfig.error
+
+final case class PredicateFailedException(message: String)
```
Comment (fthomas) [inline contrib/pureconfig/jvm/src/main/scala/eu/timepit/refined/pureconfig/error/PredicateFailedException.scala]: I was thinking about adding a special exception for refinement errors in #101 but didn't came around adding it yet to the core module. Are there any reasons against moving this later to the core module?

Comment (vlovgr) [inline contrib/pureconfig/jvm/src/main/scala/eu/timepit/refined/pureconfig/error/PredicateFailedException.scala]: Adding it to the core module sounds good, should be no reason against having it there! 👍

**Comment by vlovgr** [comment]
@fthomas Thanks for reviewing and getting this merged! 👍 

The repository readme now says `"eu.timepit" %% "refined-pureconfig" % "0.6.0"`, which is not available. Any chance of a new release? 😄

**Comment by fthomas** [comment]
@vlovgr I fixed the README. :smile:

**Pull request merged**